src/Symfony/XmlServiceMapFactory.php

```diff
@@ -85,7 +92,9 @@ public function create(): ServiceMap
 			);
 		}
 
-		return new DefaultServiceMap($services);
+		ksort($services);
+
+		return $this->serviceMap = new DefaultServiceMap($services);
```

Comment:
This doesn't make sense. This is a stateless service, it shouldn't be cached. We're not going to be asking for these objects in the extension anyway, so you can throw this away :)

Reply:
In current implementation it makes sense as those factories are used both in the extension and other places. Unless the Nette container doesn't use the same instance for dependency injection (I don't know it, you tell me 😅) it is performance optimisation, to avoid parsing same XML files multiple times (GetResponse's DI container has 2.5MB xml file, so there's no point in parsing it multiple times, in fact it would be better if params and services were parsed in the same batch, not separately).

Reply:
I think you can just ask for ParameterMap or ServiceMap in the constructor and it will work. I did that during my SymfonyCon talk.

Reply:
I did not know it, will try it, but I believe it does not change the fact this improvement is needed. If Nette calls the factory every time `ParameterMap` or `ServiceMap` is requested on DI, it should process the file only once, then reuse the result. I will check it soon 🙂.

Reply:
As any sensible DI container, nette/di keeps the services and reuses them once created.

Reply:
I removed state from the factories, but kept sorting.